### PR TITLE
[Fuzzing] add cifuzz

### PR DESCRIPTION
**Pull request by 0x34d**
Add [cifuzz](https://google.github.io/oss-fuzz/getting-started/continuous-integration) for Continuous Integration of fuzzing in PR.

### CODE REVIEW

**Comment by 0x34d** [comment]
```log
-- Could NOT find hwloc (missing: hwloc_LIBRARY hwloc_INCLUDE_DIR) 
-- Could NOT find jemalloc (missing: jemalloc_LIBRARY jemalloc_INCLUDE_DIR) 
CMake Error at /usr/local/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake-3.24/Modules/FindPkgConfig.cmake:99 (find_package_handle_standard_args)
  cmake/FindLuaJIT.cmake:32 (find_package)
  cmake/AutoOptionHelpers.cmake:44 (find_package)
  cmake/AutoOptionHelpers.cmake:116 (_check_package_depends)
  CMakeLists.txt:61 (auto_option)


-- Configuring incomplete, errors occurred!
See also "/src/trafficserver/build/CMakeFiles/CMakeOutput.log".
2023-09-19 06:45:44,680 - root - ERROR - Building fuzzers failed.
2023-09-19 06:45:44,680 - root - ERROR - Error building fuzzers for (commit: 75cbef8f8286658ad586fb9615b46847eed5891b, pr_ref: refs/pull/10462/merge).
```

There were some changes in the main code.

**Comment by 0x34d** [comment]
Why `Cifuzz` did not work?

**Comment by 0x34d** [comment]
@shukitchan can you merge?

**Comment by shukitchan** [comment]
A couple points to discuss:
1) In this early stage, we don't want the PR to be unmergeable if it is failing the cifuzz. Are you ok with that? We can check later on if we want to change that or not
2) The run took quite a while and it may be taking quite some resources. I am under the impression that all ASF shares some resources for running github actions and we may not want to run an action that took too much resources. I am checking with infrastructure team first.

**Comment by 0x34d** [comment]
> 1. In this early stage, we don't want the PR to be unmergeable if it is failing the cifuzz. Are you ok with that? We can check later on if we want to change that or not

I think this is very normal; you can set aside the build failure. If there is an actual bug in the PR, you can try to fix it.
On that note, I think I have to make some changes in `build.sh`.


> The run took quite a while and it may be taking quite some resources. I am under the impression that all ASF shares some resources for running github actions and we may not want to run an action that took too much resources. I am checking with infrastructure team first.

Yes, that's true. If you want then I can change the three sanitizers to only one `ASAN`, which is more important.

**Comment by shukitchan** [comment]
I was told that in other ASF projects with github actions, their PRs has to wait till a runner is available. 
Perhaps we don't start off running these for every PRs? How about we start running them just every week to begin with?

**Comment by 0x34d** [comment]
> I was told that in other ASF projects with github actions, their PRs has to wait till a runner is available. Perhaps we don't start off running these for every PRs? How about we start running them just every week to begin with?

This sounds really interesting.

**Comment by 0x34d** [comment]
> I was told that in other ASF projects with github actions, their PRs has to wait till a runner is available. Perhaps we don't start off running these for every PRs? How about we start running them just every week to begin with?


The core essence of cifuzz is to find bugs in PRs before they are merged.
I will ask.

File: .github/workflows/cifuzz.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: CIFuzz
+on: 
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0'  # Run at 00:00 every Sunday
```
Comment (0x34d) [inline .github/workflows/cifuzz.yml:5]: is this good @shukitchan

Comment (shukitchan) [reply .github/workflows/cifuzz.yml:5]: I think it is still run on "pull_request".

Comment (shukitchan) [reply .github/workflows/cifuzz.yml:5]: in addition to the added schedule

**Comment by shukitchan** [comment]
Also it is failing "clang-format" now. That's because there is a trailing whitespace somewhere in your latest change.

**Comment by shukitchan** [comment]
I think we can give it a go.

**Pull request merged**